### ISSUE TITLE
Error when using dice to generate tokens

### ISSUE THREAD

**Issue by Quercuas** [spetzel2020/quick-encounters#78]
First off, this is an excellent module that made my life a bit easier.
Also thank you for keeping it up to date as possible.

Unfortunately I encounter a bug when I set up an QE using the dice as the number of monsters. (ie. 1d4+2 or just 1d4)
When I try to run the encounter the usual popup happens:
![afbeelding](https://user-images.githubusercontent.com/96662470/151796341-811e0255-167f-4f05-b517-87484c1f4338.png)

But it only spawns 1 monster and gives the following error:
(I tried multiple times with various dice, but each and every iteration just spawns 1)

```
Roll#evaluate is becoming asynchronous. In the short term you may pass async=true or async=false to evaluation options to nominate your preferred behavior.
evaluate @ foundry.js:6759
getNumActors @ QuickEncounter.js:1030
generateFullExtractedActorTokenData @ QuickEncounter.js:1086
async function (async)
generateFullExtractedActorTokenData @ QuickEncounter.js:1083
run @ QuickEncounter.js:909
runAddOrCreate @ QuickEncounter.js:485
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2

```

When using a fixed number it works as expected.

**Comment by spetzel2020** [comment]
Thanks for the report! Which version of Foundry are you using? I know
random dice generation is in my standard testing but it's possible i
screwed something up when i tested for Foundry 9.

**Comment by Quercuas** [comment]
Ah of course.
I'm using 9 build 245
But I also encountered this issue in build 242

**Comment by spetzel2020** [comment]
Well I thought I had tested in 242 but clearly not well enough. I am
traveling at the moment but I will see what I can find.

**Comment by spetzel2020** [comment]
Reproduced in 9.245; only creates one token on the Scene and in the CT.  Also is not deleting that token if you select Delete All after combat.

**Comment by spetzel2020** [comment]
Verified that adding the `async:false` flag to the `evaluate()` calls in `QuickEncounters.js#getNumActors()` fixes the generation problem:
```
                if (options?.rollType === "full") {
                    r.evaluate({async: false});
                } else {//template or other
                    r.evaluate({minimize: false, maximize: true, async: false});
                }
  ```

The deletion bug seems to be unrelated. I will see if that's easy to fix, other will release fix for this Issue #78 .

**Comment by spetzel2020** [comment]
Confirmed that the deletion is not a bug; the Delete Tokens after Combat only deletes Hostile tokens and my test was using Neutral tokens; when some were converted to Hostile those ones were correctly deleted.

**Comment by spetzel2020** [comment]
Fixed in 0.9.8

**Comment by spetzel2020** [comment]
Release in 0.9.8 on the 0.9-maintenance branch. Please update as normal using https://github.com/spetzel2020/quick-encounters/releases/latest/download/module.json
